andela/bp-esa-backend
# #165676817 Use partner details from api-staging in mock placement data

#### What does this PR do?
Use retrieved partners from api-staging in mock placement data

#### Description of Task to be completed?
- implement functionality to getAndelaPartners from staging api
- save partners to redis to be retrieved next time
- replace mock generated client_name and client_id in mock placement data with retrieved partner

#### How should this be manually tested?
1. Start the application
2. Wait for automation to be carried out
3. Check the slackAutomation table to observe the partner channel-names correspond to partner names in automation records
4. Check the partner table to confirm that several partners were saved

#### Any background context you want to provide?
Client names in mock placement data are being generated randomly with a single hardcoded client_id. Owing to this:
1. Partner table is populated with a single partner since the id remain the same, even though the partner names used in slack automations are different, causing a lot of confusion
2. Uncountable number of slack channels are being created for each random partner name generated, thereby overloading the channel-list
3. Offboarding slack automations hardly succeed due to absence of channel name which matches the randomly generated partner name

#### What are the relevant pivotal tracker stories?
[#165676817](https://www.pivotaltracker.com/story/show/165676817)
